.github/workflows/setup-xapi-environment/action.yml

```diff
@@ -0,0 +1,59 @@
+name: Setup XenAPI environment
+description: Setup a XenAPI environment for making opam calls.
+
+inputs:
+  xapi_version:
+    description: "XenAPI version, pass to configure as --xapi_version=<value>"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Free space
+      shell: bash
+      run: sudo rm -rf /usr/local/lib/android
+
+    - name: Pull configuration from xs-opam
+      shell: bash
+      run: |
+        curl --fail --silent https://raw.githubusercontent.com/xapi-project/xs-opam/master/tools/xs-opam-ci.env | cut -f2 -d " " > .env
+
+    - name: Download XE_SR_ERRORCODES.xml
```

Comment:
This step is not needed in all the the workflows, why not do the step explicitly instead?

Reply:
Mistake on my end. This is only needed for SDK gen, right?

I should move it to the reusable workflow instead of this action

Reply:
guess it was needed for OCaml tests too, should I re-add it to the composite action or call it manually?

My idea behind `setup-xapi-environment` was to have one reusable action for any instance so if it's used by both workflows I'd add the wget to the action - thoughts?